src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -2367,6 +2380,16 @@ private void doSignalWriteIdleTimeoutIfNecessary()
             }
         }
 
+        private void doSignalCloseTimeout()
+        {
+            closeTimeoutAt = System.currentTimeMillis() + CLOSE_TIMEOUT;
+
+            if (closeTimeoutId == NO_CANCEL_ID)
+            {
+                closeTimeoutId = signaler.signalAt(closeTimeoutAt, routeId, replyId, CLOSE_SIGNAL_ID);
+            }
```

Comment:
Under what conditions do we need to make this check?
If it is expected to always be true it can be changed to an assert.